[Proxy] IndexOutOfBoundsException: Index: 2, Size: 2 when connecting with DataGrip due to metadata quer

### Which version of ShardingSphere did you use?
DBMS: MySQL (ver. 5.7.22-ShardingSphere-Proxy 5.5.2)
Case sensitivity: plain=exact, delimited=exact
Driver: MySQL Connector/J (ver. mysql-connector-java-8.0.25 (Revision: 08be9e9b4cba6aa115f9b27b215887af40b159e0), JDBC4.2)
Effective version: MySQL (ver. 5.7.25)

Ping: 1 sec, 188 ms
SSL: no

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
The proxy should correctly handle the metadata query sent by DataGrip and return the result set with the correct number of columns (3 columns), or handle the function calls gracefully without throwing an exception.

### Actual behavior
When connecting to ShardingSphere Proxy using **JetBrains DataGrip (2025.3)**, an `IndexOutOfBoundsException` occurs in `DatabaseAdminQueryBackendHandler`.

The error `Index: 2, Size: 2` suggests that DataGrip is requesting **3 columns**, but the internal logic in `DatabaseAdminQueryBackendHandler` (or the underlying `AbstractMemoryQueryResult`) only prepared **2 columns**, causing a crash when accessing the 3rd column.

### Reason analyze (If you can)
DataGrip sends the following SQL to check the current session status (database, schema, and user):

```sql
/* ApplicationName=DataGrip 2025.3 */ 
select database(), schema(), left(user(),instr(concat(user(),'@'),'@')-1)
```

This query requests **3 expressions**:
            1. `database()`
            2. `schema()`
            3. `left(user()...)`

    The stack trace indicates the error happens at `DatabaseAdminQueryBackendHandler.getRowData(DatabaseAdminQueryBackendHandler.java:82)`. It seems the result row construction for this specific admin query does not match the column count expected by the execution engine.

            ### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
            1. Start ShardingSphere Proxy (MySQL Protocol).
            2. Connect to the proxy using **JetBrains DataGrip** (e.g., version 2024.x or 2025.3).
            3. Observe the proxy logs or DataGrip console.


            ### Stack trace of logs
```plain text
[ERROR] 2025-12-04 05:44:36.572 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 

java.lang.IndexOutOfBoundsException: Index: 2, Size: 2

	at java.base/java.util.LinkedList.checkElementIndex(LinkedList.java:566)

	at java.base/java.util.LinkedList.get(LinkedList.java:487)

	at org.apache.shardingsphere.infra.executor.sql.execute.result.query.type.memory.AbstractMemoryQueryResult.getValue(AbstractMemoryQueryResult.java:76)

	at org.apache.shardingsphere.infra.merge.result.impl.transparent.TransparentMergedResult.getValue(TransparentMergedResult.java:44)

	at org.apache.shardingsphere.proxy.backend.handler.admin.DatabaseAdminQueryBackendHandler.getRowData(DatabaseAdminQueryBackendHandler.java:82)

	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.getQueryRowPacket(MySQLComQueryPacketExecutor.java:130)

	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.getQueryRowPacket(MySQLComQueryPacketExecutor.java:53)

	at org.apache.shardingsphere.proxy.frontend.mysql.command.MySQLCommandExecuteEngine.writeQueryData(MySQLCommandExecuteEngine.java:78)

	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.doExecuteCommand(CommandExecutorTask.java:132)

	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:121)

	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)

	at com.alibaba.ttl.TtlRunnable.run(TtlRunnable.java:60)

	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)

	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)

	at java.base/java.lang.Thread.run(Thread.java:1583)
```
    at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
    at com.alibaba.ttl.TtlRunnable.run(TtlRunnable.java:60)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.base/java.lang.Thread.run(Thread.java:1583)
         ```

--- a/proxy/backend/dialect/mysql/src/main/java/org/apache/shardingsphere/proxy/backend/mysql/handler/admin/factory/withoutfrom/MySQLSelectWithoutFromAdminExecutorFactory.java
+++ b/proxy/backend/dialect/mysql/src/main/java/org/apache/shardingsphere/proxy/backend/mysql/handler/admin/factory/withoutfrom/MySQLSelectWithoutFromAdminExecutorFactory.java
@@ -93,7 +93,15 @@ public final class MySQLSelectWithoutFromAdminExecutorFactory {
             return Optional.of(new NoResourceShowExecutor(sqlStatement));
         }
         boolean isUseDatabase = null != databaseName || sqlStatement.getFrom().isPresent();
+        if (!isUseDatabase && hasMultipleProjections(sqlStatement)) {
+            return Optional.empty();
+        }
         return isUseDatabase ? Optional.empty() : Optional.of(new UnicastResourceShowExecutor(sqlStatement, sql));
+    }
+    
+    private static boolean hasMultipleProjections(final SelectStatement sqlStatement) {
+        Collection<ProjectionSegment> projections = sqlStatement.getProjections().getProjections();
+        return projections.size() > 1;
     }
     
     private static boolean isEmptyResource(final ShardingSphereMetaData metaData) {

--- a/proxy/backend/dialect/mysql/src/test/java/org/apache/shardingsphere/proxy/backend/mysql/handler/admin/MySQLAdminExecutorCreatorTest.java
+++ b/proxy/backend/dialect/mysql/src/test/java/org/apache/shardingsphere/proxy/backend/mysql/handler/admin/MySQLAdminExecutorCreatorTest.java
@@ -73,6 +73,7 @@ import org.apache.shardingsphere.test.infra.framework.extension.mock.StaticMockS
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Optional;
@@ -404,5 +405,49 @@ class MySQLAdminExecutorCreatorTest {
         SQLStatementContext sqlStatementContext = new CommonSQLStatementContext(new DeleteStatement(databaseType));
         Optional<DatabaseAdminExecutor> actual = new MySQLAdminExecutorCreator().create(sqlStatementContext, "DELETE FROM t", "", Collections.emptyList());
         assertThat(actual, is(Optional.empty()));
+    }
+    
+    @Test
+    void assertCreateWithNoFromAndMultiProjectionsSkipsAdmin() {
+        ResourceMetaData resourceMetaData = new ResourceMetaData(Collections.singletonMap("ds_0", new MockedDataSource()));
+        ShardingSphereDatabase database = new ShardingSphereDatabase("db_0", databaseType, resourceMetaData, mock(RuleMetaData.class), Collections.emptyList());
+        initProxyContext(Collections.singleton(database));
+        
+        SelectStatement selectStatement = mock(SelectStatement.class);
+        when(selectStatement.getFrom()).thenReturn(Optional.empty());
+        ProjectionsSegment projectionsSegment = mock(ProjectionsSegment.class);
+        when(projectionsSegment.getProjections()).thenReturn(Arrays.asList(
+                new ExpressionProjectionSegment(0, 10, "database()"),
+                new ExpressionProjectionSegment(0, 10, "schema()"),
+                new ExpressionProjectionSegment(0, 10, "left(user(),instr(concat(user(),'@'),'@')-1)")));
+        when(selectStatement.getProjections()).thenReturn(projectionsSegment);
+        SelectStatementContext sqlStatementContext = mock(SelectStatementContext.class);
+        when(sqlStatementContext.getSqlStatement()).thenReturn(selectStatement);
+        
+        Optional<DatabaseAdminExecutor> actual =
+                new MySQLAdminExecutorCreator().create(sqlStatementContext, "SELECT database(),schema(),left(user(),instr(concat(user(),'@'),'@')-1)", null, Collections.emptyList());
+        assertThat(actual, is(Optional.empty()));
+    }
+    
+    @Test
+    void assertCreateWithMultiSystemVariablesUseSysVarExecutor() {
+        initProxyContext(Collections.emptyList());
+        SelectStatement selectStatement = mock(SelectStatement.class);
+        when(selectStatement.getFrom()).thenReturn(Optional.empty());
+        ProjectionsSegment projectionsSegment = mock(ProjectionsSegment.class);
+        VariableSegment v1 = new VariableSegment(0, 0, "version", "SESSION");
+        VariableSegment v2 = new VariableSegment(0, 0, "transaction_isolation", "SESSION");
+        when(projectionsSegment.getProjections()).thenReturn(Arrays.asList(
+                new ExpressionProjectionSegment(0, 10, "@@session.version", v1),
+                new ExpressionProjectionSegment(0, 10, "@@session.transaction_isolation", v2)));
+        when(selectStatement.getProjections()).thenReturn(projectionsSegment);
+        
+        SelectStatementContext sqlStatementContext = mock(SelectStatementContext.class);
+        when(sqlStatementContext.getSqlStatement()).thenReturn(selectStatement);
+        
+        Optional<DatabaseAdminExecutor> actual = new MySQLAdminExecutorCreator().create(
+                sqlStatementContext, "SELECT @@session.version, @@session.transaction_isolation", null, Collections.emptyList());
+        assertTrue(actual.isPresent());
+        assertThat(actual.get(), isA(MySQLSystemVariableQueryExecutor.class));
     }
 }

